### DISCUSSION TITLE
Include a few Zigbee examples as reference sample applications/projects?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#34] [Ideas]
An idea could be to include a few Zigbee examples as reference application samples similar most commercial Zigbee SDKs, and since some of those samples are open souce it might be a good idea to copy and/or adapt those examples for reference that help with ideas. Check out:

https://github.com/espressif/esp-zigbee-sdk/tree/main/examples

https://github.com/SiliconLabs/zigbee_applications

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/zigbee.html

https://software-dl.ti.com/simplelink/esd/plugins/simplelink_zigbee_sdk_plugin/1.60.01.09/exports/docs/zigbee_user_guide/html/zigbee/developing_zigbee_applications/example_applications.html

**Comment by Hedda** [comment]
Not specifically there but FYI, a basic but common generic example sample application/project seen many asking for is an always-on Zigbee Router (ZR) device application for used in dedicated Zigbee signal repeater / Zigbee range extender or as part of a other project,

I guess it would need be configured to automatically always try to join a Zigbee network for 60-seconds when power-on if it is not already joined to a Zigbee network + might also need way to reset devices with no button?

For example check out their "Zigbee Human Detection" example (functions as an always-on Zigbee Router device with join button):

https://github.com/SiliconLabs/zigbee_applications/tree/master/zigbee_human_detection

"The sample application provided is for ZigBee Router as a range extender. The router joins the coordinator and enables permit join. It then only handles the stack callbacks."

https://www.silabs.com/documents/login/user-guides/RS9113-ZigBee-Application-UserGuide.pdf

While not a must, if someone does make a such generic always-on Zigbee Router device application then it would be awesome if also make the the transmit power setting configurable over Zigbee in the same way that Koenkk's Z-Stack Router firmware is, see:

https://github.com/Koenkk/Z-Stack-firmware/issues/341

That is, set the default transmit power to 9dBm but expose an optional attribute to change TX power up to 20 dBm via Zigbee:

https://github.com/Koenkk/Z-Stack-firmware/blob/master/router/Z-Stack_3.x.0/CHANGELOG.md

https://github.com/Koenkk/Z-Stack-firmware/blob/master/router/Z-Stack_3.x.0/firmware.patch

![image](https://github.com/darkxst/silabs-firmware-builder/assets/6320001/d5fe612c-05cd-4850-a5c9-b4186ea4d7e8)

**Comment by ffenix113** [comment]
Hello Hedda,

This is indeed something that this project needs, and it does make great sense!
I think good idea would be to add an example code in time when adding a new feature/cluster. 

Unfortunately, I need to split my time between different things that needs to be done in this project, as it is in active stage of being developed, which means that some things might take longer to be addressed. I.e. for now I am finishing up initial implementation for contact sensor, and then will update some documentation, and will move on to battery status and soil moisture sensor. All of this steps will include adding examples for sure.

Thank you for the suggestion! 🚀
